diff --git b/js/form.js a/js/form.js
index 526bc4e..cfb7ab1 100644
--- b/js/form.js
+++ a/js/form.js
@@ -37,6 +37,7 @@ class Form {
 
     handleReviewSubmission(e) {
         if (this.validateForm.call(this)) {
+            console.log('submitting');
             const review = this.getReviewFromForm();
             this.postReview(review)
                 .then(this.onFormSubmissionSuccess.bind(this))
@@ -56,7 +57,9 @@ class Form {
     }
 
     onFormSubmissionSuccess(res) {
+        console.log({ res });
         if (res.status === 201) {
+            console.log('Post successful');
 
             let toast = VanillaToasts.create({
                 title: 'Review submitted!',
@@ -123,6 +126,7 @@ class Form {
     }
 
     saveReviewToIDB(review) {
+        console.log('saving review', { review });
         const objStoreName = 'reviews';
         this.IDB.set(undefined, review, objStoreName);
     }
@@ -130,6 +134,7 @@ class Form {
     async getReviewsFromIDB() {
         const objStoreName = 'reviews';
         const reviews = await this.IDB.getAll(objStoreName);
+        console.log('all review form submissions', { reviews });
         return reviews;
     }
 
@@ -143,6 +148,7 @@ class Form {
 
         function postReviewWithinIDB(review, key) {
             postReview(review).then(res => {
+                console.log({ res });
                 if (res.status === 201) {
                     idbDelete(key, objStoreName);
                     fetchReviewsFromNetwork();
@@ -152,6 +158,7 @@ class Form {
     }
 
     init() {
+        console.log('Initializing form');
         this.initElements();
         this.initHandlers();
         this.setRestaurantIdToForm();
